File: v1.1/src/examples/psibaby.m
main()
{
   array a1^2 <10 10>;
   array fish^1 <10>;
   array goat^1 <10>;
   array duck^1 <10>;
   array ind^2 <10 1>; 
   array in0^2 <10 1>;
   array in1^2 <10 1>;
   array x^1 <1>;
   const array frog^1 <1> = <5>;

#   fish = <5> psi a1;
   goat = x drop (x take duck);
   goat = frog drop (frog take duck);
   fish = <5> drop (<5> take duck);
   
#   goat = <5> psi omega <1 1> a1;
#  duck = (((iota <10 1>) * omega <1 1> <1 1>) + omega <1 1> <0 4>) psi omega <1 2> a1;
#   ind = iota <10 1>;
#   in0 = ind * omega < 1 1> <1 1>;
#   in1 = in0 + omega <1 1> <0 4>;
#   duck = in1 psi omega <1 2> a1;
}

File: v1.1/src/examples/psibaby.c
#include <stdlib.h>
#include "moalib.e"

main()

{
  double *offset0;
  int i0;
  double *shift;
  double _frog[]={5.000000};
  double _x[1];
  double _goat[10];
  double _duck[10];

/*******
goat=x drop (x take duck)
********/

  shift=_goat+0;
  offset0=_duck+((((int) _x[0])+10)*(((((int) _x[0]))<0))+(((int) _x[0]))*(((((int) _x[0]))>0)));
  for (i0=0; i0<((abs(((int) _x[0])))-(abs(((int) _x[0])))); i0++) {
    *(shift)= *(offset0);
    offset0+=1;
    shift+=1;
  }


/*******
goat=frog drop (frog take duck)
********/



/*******
fish=<5.000000> drop (<5.000000> take duck)
********/



}


